TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -193,14 +179,28 @@ internal void InitCopyDependencyWatchers()
         /// Start continuous background compilation on a newly opened file
         /// </summary>
         /// <param name="docContext">The Document context</param>
+        /// <param name="projectKey">Project's Key</param>
         /// <param name="sourceText">The source text</param>
+        /// <param name="copyFolders">List of copy folders associated to the project</param>
         /// <returns>The corresponding FileCompiler instance.</returns>
-        public FileCompiler OpenTextDocument(DocumentContext docContext, string sourceText) => OpenTextDocument(docContext, sourceText, LsrTestOptions);
+        public FileCompiler OpenTextDocument(DocumentContext docContext, string projectKey, string sourceText, List<string> copyFolders) => OpenTextDocument(docContext, projectKey, sourceText, copyFolders, LsrTestOptions);
 
-        private FileCompiler OpenTextDocument(DocumentContext docContext, string sourceText, LsrTestingOptions lsrOptions)
+        /// <summary>
+        /// Bind a document to a FileCompiler instance and update its content for a parsing..
+        /// </summary>
+        /// <param name="docContext">The document context to bind</param>
+        /// <param name="sourceText">The sourcre text of the document</param>
+        /// <param name="lsrOptions">LSR options</param>
+        /// <returns>The FileCompiler instance associated to the document context</returns>
+        internal FileCompiler BindFileCompilerSourceTextDocument(DocumentContext docContext, string sourceText, LsrTestingOptions lsrOptions)
```

Comment:
Because of a hack in TypeCobolServer.OnDidChangeTextDocument to rebind an opened document to a new FileCompiler instance an thus to refresh the document if whole document content has changed.

Reply:
Yes, ok we can keep it